api/src/models/course.js

```diff
@@ -1,8 +1,98 @@
 const mongoose = require("mongoose");
 const uniqueValidator = require("mongoose-unique-validator");
-
+/**
+ * @swagger
+ *  components:
+ *    $ref: './regulation.js/#components'
+ *    schemas:
+ *      Course:
+ *        type: object
+ *        required:
+ *          - courseName
+ *          - courseCode
```

Comment:
Maybe we should keep this as `name` and `code` rather than `courseName` and `courseCode`